src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/InstalledAndTransitivePackageFeed.cs

```diff
@@ -25,59 +25,51 @@ public InstalledAndTransitivePackageFeed(IEnumerable<PackageCollectionItem> inst
         /// <inheritdoc cref="IPackageFeed.ContinueSearchAsync(ContinuationToken, CancellationToken)" />
         public override async Task<SearchResult<IPackageSearchMetadata>> ContinueSearchAsync(ContinuationToken continuationToken, CancellationToken cancellationToken)
         {
-            var searchToken = continuationToken as FeedSearchContinuationToken;
-            if (searchToken == null)
-            {
-                throw new InvalidOperationException(Strings.Exception_InvalidContinuationToken);
-            }
+            var searchToken = ThrowIfNotFeedSearchContinuationToken(continuationToken);
+            cancellationToken.ThrowIfCancellationRequested();
 
             // Remove transitive packages from project references
             IEnumerable<PackageCollectionItem> pkgsWithOrigins = _transitivePackages
                 .Where(t => t.PackageReferences.Any(x => x is ITransitivePackageReferenceContextInfo y && y.TransitiveOrigins.Any()));
 
-            IEnumerable<PackageCollectionItem> pkgs = _installedPackages.Concat(pkgsWithOrigins);
-            PackageCollectionItem[] allPkgs = PerformLookup(pkgs, searchToken);
+            IPackageSearchMetadata[] intalledItems = await DoSearchAsync(_installedPackages, searchToken, cancellationToken);
```

Comment:
typo in variable name

```suggestion
            IPackageSearchMetadata[] installedItems = await DoSearchAsync(_installedPackages, searchToken, cancellationToken);
```